Adding bloom command meta data, bloom group and bloom data type documentaion (#233)
valkey-io/valkey-doc: commands/bf.insert.md

> Creates a bloom object with the specified parameters. If a parameter is not specified then the default value will be used. If ITEMS is specified then it will also attempt to add all items specified after

Author: ITEMS isn't required for us for INSERT. But if the user does specify ITEMS they do need to provide at least one item

> Creates a bloom object with the specified parameters. If a parameter is not specified then the default value will be used. If ITEMS is specified then it will also attempt to add all items specified after

Author: Items isn't required for us, so BF.INSERT could create an empty bloom filter. Also updated sentence to read more clearly

* SEED (optional) - The seed the hash functions will use
* NONSCALING (optional) - Will make it so the filter can not scale
* VALIDATESCALETO validatescaleto (optional) - Checks if the filter could scale to this capacity and if not show an error and don’t create the bloom filter
* ITEMS (optional) - Items we will add to the bloom filter

## Examples
```
127.0.0.1:6379> BF.INSERT key ITEMS item1 item2
1) (integer) 1
2) (integer) 1
# This does not update the capcity but uses the origianl filters values
127.0.0.1:6379> BF.INSERT key CAPACITY 1000 ITEMS item2 item3
1) (integer) 0
2) (integer) 1
127.0.0.1:6379> BF.INSERT key_new CAPACITY 1000
[]
```

```
127.0.0.1:6379> BF.INSERT key NONSCALING VALIDATESCALETO 100
cannot use NONSCALING and VALIDATESCALETO options together
127.0.0.1:6379> BF.INSERT key CAPACITY 1000  VALIDATESCALETO 999999999999999999999 ITEMS item2 item3
provided VALIDATESCALETO causes bloom object to exceed memory limit
127.0.0.1:6379> BF.INSERT key VALIDATESCALETO 999999999999999999999 EXPANSION 1 ITEMS item2 item3
provided VALIDATESCALETO causes false positive to degrade to 0
```
```
127.0.0.1:6379> BF.INSERT key NOCREATE ITEMS item1 item2

> not found

Author: You're right is an error updated to show the full error as well

Creates a bloom filter with the specified parameters. If a parameter is not specified then the default value will be used. If ITEMS is specified then it will also attempt to add all items specified

> ## Insert Fields

Reviewer: Some of the options below are missing the "``"

Author: Fixed in newest commit

* CAPACITY capacity -  capacity for the initial bloom filter
* ERROR `fp_error` - The false positive rate for the bloom filter

> * EXPANSION expansion - The expansion rate for a scaling filter

Reviewer: Can we use the same wording that we use in BF.RESERVE?

* NOCREATE  - Will not create the bloom filter and add items if the filter does not exist already
* TIGHTENING `tightening_ratio` - The tightening ratio for the bloom filter

> * SEED seed - The seed the hash functions will use

Reviewer: TODO: Add more wording

> * NONSCALING - Will make it so the filter can not scale

Reviewer: Follow wording from BF.RESERVE

> * VALIDATESCALETO `validatescaleto` - Checks if the filter could scale to this capacity and if not show an error and don’t create the bloom filter

Reviewer: ```suggestion
* VALIDATESCALETO `validatescaleto` - Validates if the filter can scale out and reach to this capacity based on limits and if not, return an error without creating the bloom filter
```

> * ITEMS item - One or more items we will add to the bloom filter

Reviewer: ```suggestion
* ITEMS item - One or more items to be added to the bloom filter
```

Due to the nature of  NONSCALING and VALIDATESCALETO arguments, specifying NONSCALING and VALIDATESCALETO isn't allowed

## Examples

```
127.0.0.1:6379> BF.INSERT key ITEMS item1 item2
1) (integer) 1
2) (integer) 1

> # This does not update the capcity but uses the origianl filters values

Reviewer: ```suggestion
# This does not update the capacity since the filter already exists. It only adds the provided items.
```

If the bloom filter does not exist under the specified name, a bloom filter is created with the specified parameters. Default properties will be used if the options below are not specified.

> When the ITEMS option is provided, all items provided will be attempted to be added.

Reviewer: ```suggestion
When the `ITEMS` option is provided, all items provided will be attempted to be added.
```

* ITEMS *item* - One or more items to be added to the bloom filter.

Due to the nature of `NONSCALING` and `VALIDATESCALETO` arguments, specifying `NONSCALING` and `VALIDATESCALETO` together is not allowed.

## Examples

```
127.0.0.1:6379> BF.INSERT key ITEMS item1 item2
1) (integer) 1
2) (integer) 1
# This does not update the capacity since the filter already exists. It only adds the provided items.
127.0.0.1:6379> BF.INSERT key CAPACITY 1000 ITEMS item2 item3
1) (integer) 0
2) (integer) 1
127.0.0.1:6379> BF.INSERT key_new CAPACITY 1000
[]
```

```
127.0.0.1:6379> BF.INSERT key NONSCALING VALIDATESCALETO 100
(error) ERR cannot use NONSCALING and VALIDATESCALETO options together
127.0.0.1:6379> BF.INSERT key CAPACITY 1000  VALIDATESCALETO 999999999999999999 ITEMS item2 item3
(error) ERR provided VALIDATESCALETO causes bloom object to exceed memory limit
127.0.0.1:6379> BF.INSERT key VALIDATESCALETO 999999999999999999 EXPANSION 1 ITEMS item2 item3
(error) ERR provided VALIDATESCALETO causes false positive to degrade to 0
```
```
127.0.0.1:6379> BF.INSERT key NOCREATE ITEMS item1 item2

> (error) ERR not found

Reviewer 2: This is not a very good error, is it too late to make it better?

Author: This error message is from existing API and error messages from existing client libraries that support bloom filters
We followed the existing error messages to be API compatible with the bloom filter commands of existing client libraries
